test/DeviceStateTrackerSpec.ts

```diff
@@ -0,0 +1,140 @@
+import DeviceStateTracker from '../src/lib/DeviceStateTracker';
+import DeviceStateEvent from '../src/lib/DeviceStateEvent';
+// Uncomment to test with JS configMock
+// import { mediaQueries, DeviceState, WrongDeviceState } from './configMockJs';
+import { mediaQueries, DeviceState, WrongDeviceState } from './configMock';
+import { expect } from 'chai';
+import {} from 'mocha';
+const matchMediaMock = require('match-media-mock').create();
+
+// Use matchMediaMock instead of window.matchMedia native
+window.matchMedia = <any> matchMediaMock;
+
+/**
+ * Small helper function for populating media key of QueryList object
+ * Handles JavaScript and TypeScript 'enums'
+ * @param deviceState
+ * @param deviceStateTracker
+ */
+const populateQueryListMediaKey = (deviceStateTracker) => {
+	const enumValues = Object.keys(DeviceState)
+		.reduce(
+			(accumulator:Array<string>, value) => {
+				if (isNaN(parseInt(value, 10))) {
+					accumulator.push(value);
+				}
+
+				return accumulator;
+			},
+			[]);
+
+	(<any> deviceStateTracker)._queryLists.forEach((mq, index) => mq.media = mediaQueries[enumValues[index]]);
```

Comment:
Yes anti-pattern, but for now the mock MediaQueryList doesn't have a media key on it's list object so I need to append this for testing. I've opened a issue on the authors github but didn't have a reply yet.